internal/gatewayapi/securitypolicy.go

```diff
@@ -608,6 +708,61 @@ func (t *Translator) translateSecurityPolicyForRoute(
 	return errs
 }
 
+func (t *Translator) translateSecurityPolicyForTCPRoute(
```

Comment:
curious why cant we reuse the existing func and API as is ?
and leverage it to build out the right xds in the next layer